### PR TITLE
feat: refactor ada form fields

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #2127

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

The initial AC in the ticket was to update the ada step in the public site application to remove our use of plain input tags and make use of our Field components instead.

We updated the public site ada step to use our FieldGroup component, and noticed that the ada fields on the partners side manual application input form were just repeating Field components instead of using a FieldGroup, so we updated that as well, and added the shared keys to the shared-helpers package.

## How Can This Be Tested/Reviewed?

Ensure that the ADA step in the public application works exactly as before: (1) The values should save appropriately (2) Checking the none value should uncheck all the other values, and checking any other value should uncheck the none value (3) The checkboxes should autofill to their previous input when navigating forward and backward between steps (4) The question should require at least one checkbox. 

The only test for the partners side should be that checkboxes save - it is not required and has no special checkbox behavior.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/pages/applications/household/ada.tsx [hunk 1]
```diff
@@ -58,11 +58,52 @@ const ApplicationAda = () => {
...
+        },
+      },
+    }
+  })
+
+  adaFeaturesOptions.push({
+    id: "no-features",
+    label: t(`t.no`),
+    value: "no-features",
+    defaultChecked: false,
```
Comment (emilyjablonski) [inline sites/public/pages/applications/household/ada.tsx:91]: TODO

File: shared-helpers/src/formKeys.ts [hunk 2]
```diff
@@ -68,6 +68,18 @@ export const contactPreferencesKeys = [
...
+    id: "mobility",
+  },
+  {
+    id: "vision",
+  },
+  {
+    id: "hearing",
+  },
+]
+
```
Comment (seanmalbert) [inline shared-helpers/src/formKeys.ts:82]: What's the benefit of this over an array of string?

File: sites/public/pages/applications/household/ada.tsx [hunk 3]
```diff
@@ -63,6 +65,50 @@ const ApplicationAda = () => {
...
+        onChange: () => {
+          setTimeout(() => {
+            setValue("app-accessibility-no-features", false)
+          }, 1)
+        },
+      },
+    }
+  })
+
+  console.log(application)
```
Comment (seanmalbert) [inline sites/public/pages/applications/household/ada.tsx:88]: remaining console log

File: sites/public/pages/applications/household/ada.tsx [hunk 4]
```diff
@@ -40,14 +40,16 @@ const ApplicationAda = () => {
...
       accessibility: {
-        mobility: data.mobility,
-        vision: data.vision,
-        hearing: data.hearing,
+        mobility: !!data["app-accessibility-mobility"],
+        vision: !!data["app-accessibility-vision"],
+        hearing: !!data["app-accessibility-hearing"],
       },
     })
+    console.log(conductor)
```
Comment (seanmalbert) [inline sites/public/pages/applications/household/ada.tsx:52]: log

**Review by seanmalbert — CHANGES_REQUESTED** [review]

**Review by seanmalbert — APPROVED** [review]
lgtm

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**